### PR TITLE
feat: Task query page & reactor manager & resolve bug

### PR DESCRIPTION
**Pull request by gavin1ee**
Task query page & reactor manager & resolve bug

### CODE REVIEW

File: src/Contracts/MASA.Scheduler.Contracts.Server/Infrastructure/Managers/BaseSchedulerManager.cs [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ﻿// Copyright (c) MASA Stack All rights reserved.
 // Licensed under the Apache License. See LICENSE.txt in the project root for license information.
 
+using Masa.Utils.Security.Cryptography;
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Infrastructure/Managers/BaseSchedulerManager.cs:4]: global using

File: src/Services/Masa.Scheduler.Services.Server/Domain/EventHandler/NotifyTaskRunResultDomainEventHandler.cs [hunk 2]
```diff
@@ -34,8 +36,8 @@ public async Task NotifyTaskRunResultAsync(NotifyTaskRunResultDomainEvent @event
             case TaskRunStatus.Success:
                 message = "Task run success";
                 break;
-            case TaskRunStatus.Stop:
-                message = "Task run stop";
+            case TaskRunStatus.Timeout:
```
Comment (MayueCif) [inline src/Services/Masa.Scheduler.Services.Server/Domain/EventHandler/NotifyTaskRunResultDomainEventHandler.cs:39]: change to :
var message = XXX switch
{
  TaskRunStatus.Success=>"",
TaskRunStatus.Timeout=>"",
}

File: src/Services/Masa.Scheduler.Services.Server/Domain/Managers/Servers/SchedulerServerManager.cs [hunk 3]
```diff
@@ -61,40 +64,34 @@ public async Task<WorkerModel> GetWorker(RoutingStrategyTypes routingType)
...
-            throw new UserFriendlyException("Worker host cannot empty");
+            return Task.FromResult(workerModel);
         }
 
-        var hostInfo = workerHost.Split(':');
-
-        if (hostInfo.Length != 2)
+        var uri = new Uri(workerHost);
+       
+        if(uri.Scheme == "http")
```
Comment (MayueCif) [inline src/Services/Masa.Scheduler.Services.Server/Domain/Managers/Servers/SchedulerServerManager.cs:78]: add SchemeConst

File: src/Services/Masa.Scheduler.Services.Server/Domain/Managers/Servers/SchedulerServerManager.cs [hunk 4]
```diff
@@ -146,11 +153,61 @@ public Task StartAssignAsync()
...
+                        {
+                            workerModel = await GetWorker(taskDto.Job.RoutingStrategy);
+                        }
+
+                        await CheckHeartbeat(workerModel!);
+
+                        if(workerModel.Status != ServiceStatus.Normal)
+                        {
+                            _data.TaskQueue.Enqueue(taskDto);
+                            await Task.Delay(1000);
```
Comment (MayueCif) [inline src/Services/Masa.Scheduler.Services.Server/Domain/Managers/Servers/SchedulerServerManager.cs:194]: ?

File: src/Services/Masa.Scheduler.Services.Server/Domain/Managers/Servers/SchedulerServerManager.cs [hunk 5]
```diff
@@ -146,11 +153,61 @@ public Task StartAssignAsync()
                     continue;
                 }
 
+                if(!_data.ServiceList.Any())
+                {
+                    await Task.Delay(1000);
```
Comment (MayueCif) [inline src/Services/Masa.Scheduler.Services.Server/Domain/Managers/Servers/SchedulerServerManager.cs:158]: ?

File: src/Web/MASA.Scheduler.Web.Admin/Pages/Teams/Components/Projects.razor.cs [hunk 6]
```diff
@@ -5,15 +5,31 @@ namespace Masa.Scheduler.Web.Admin.Pages.Teams.Components;
 public partial class Projects
 {
     [Parameter]
-    public Guid TeamId { get; set; }
+    public Guid? TeamId
+    {
+        get
+        {
+            return _teamId;
+
```
Comment (MayueCif) [inline src/Web/MASA.Scheduler.Web.Admin/Pages/Teams/Components/Projects.razor.cs:13]: delete blanl line

File: src/Web/MASA.Scheduler.Web.Admin/Pages/Teams/Components/SchedulerJobJobs.razor.cs [hunk 7]
```diff
@@ -17,31 +14,73 @@ public partial class Jobs : ProCompontentBase
         }
         set
         {
-            _project = value;
+            if (_project?.Id != value?.Id)
```
Comment (MayueCif) [inline src/Web/MASA.Scheduler.Web.Admin/Pages/Teams/Components/SchedulerJobJobs.razor.cs:17]: _project and value both null

**Review by MayueCif — CHANGES_REQUESTED** [review]
modify

**Review by MayueCif — APPROVED** [review]

**Comment by MayueCif** [comment]
@doddgu

File: src/Contracts/MASA.Scheduler.Contracts.Server/Masa.Scheduler.Contracts.Server.csproj [hunk 8]
```diff
@@ -10,6 +10,11 @@
     <PackageReference Include="FluentValidation" Version="11.0.1" />
     <PackageReference Include="Masa.Contrib.Dispatcher.IntegrationEvents.Dapr" Version="0.4.0-rc.2" />
     <PackageReference Include="Masa.Utils.Caching.Core" Version="0.4.0-rc1" />
+    <PackageReference Include="Masa.Utils.Security.Cryptography" Version="0.4.0" />
+    <PackageReference Include="Microsoft.AspNetCore.SignalR.Client" Version="6.0.5" />
+    <PackageReference Include="Microsoft.AspNetCore.SignalR.Client.Core" Version="6.0.5" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.5" />
```
Comment (doddgu) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Masa.Scheduler.Contracts.Server.csproj:16]: Why contacts depend on EF?

File: src/Web/MASA.Scheduler.Web.Admin/Masa.Scheduler.Web.Admin.csproj [hunk 9]
```diff
@@ -21,11 +21,23 @@
...
-    <ProjectReference Include="..\..\ApiGateways\Caller\MASA.Scheduler.ApiGateways.Caller\Masa.Scheduler.ApiGateways.Caller.csproj" />
+    <ProjectReference Include="..\..\ApiGateways\Caller\Masa.Scheduler.ApiGateways.Caller\Masa.Scheduler.ApiGateways.Caller.csproj" />
   </ItemGroup>
 	
   <ItemGroup>
     <Folder Include="Components\" />
   </ItemGroup>
+	
+  <ItemGroup>
+    <Content Update="C:\Users\liguanyao\.nuget\packages\masa.blazor.experimental.components\0.2.6\contentFiles\any\net6.0\wwwroot\css\experimental.css">
```
Comment (doddgu) [inline src/Web/MASA.Scheduler.Web.Admin/Masa.Scheduler.Web.Admin.csproj:32]: `C:\Users\liguanyao\`?

File: src/Web/MASA.Scheduler.Web.Admin/Pages/Teams/Components/SchedulerJobJobs.razor.cs [hunk 10]
```diff
@@ -57,12 +62,127 @@ public partial class Jobs : ProCompontentBase
...
+    {
+        get => _queryStatus;
+        set
+        {
+            if (_queryStatus != value)
+            {   
+                _queryStatus = value;
+                OnQueryDataChanged();
+            }
+
```
Comment (doddgu) [inline src/Web/MASA.Scheduler.Web.Admin/Pages/Teams/Components/SchedulerJobJobs.razor.cs:78]: remove the blank line

File: src/Web/MASA.Scheduler.Web.Admin/Pages/Teams/Components/SchedulerJobJobs.razor.cs [hunk 11]
```diff
@@ -57,12 +62,127 @@ public partial class Jobs : ProCompontentBase
...
+    {
+        get => _queryJobName;
+        set
+        {
+            if (_queryJobName != value)
+            {
+                _queryJobName = value;
+                OnQueryDataChanged();
+            }
+
```
Comment (doddgu) [inline src/Web/MASA.Scheduler.Web.Admin/Pages/Teams/Components/SchedulerJobJobs.razor.cs:92]: remove the blank line

**Review by doddgu — CHANGES_REQUESTED** [review]

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**